### PR TITLE
Delete Poll Functionality Fix

### PR DESCRIPTION
**Pull request by alexyoung23j**
## Link to Issue
Closes: #3878 (sorta)

## Description of Changes
- The delete poll functionality was removed from the site (@zakhap confirmed this was not intentional). The delete poll route was also not working so I have fixed that. 

## Test Plan
- Tested locally. 

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

File: packages/commonwealth/server/routes/deletePoll.ts [hunk 1]
```diff
@@ -69,10 +69,19 @@ const deletePoll = async (
...
     if (!poll) {
       throw new AppError(Errors.NoPoll);
     }
+
+    const votes = await models.Vote.findAll({
+      where: {
+        poll_id: poll.id,
+      },
+    });
+    await Promise.all(votes.map((v) => v.destroy()));
```
Comment (jnaviask) [inline packages/commonwealth/server/routes/deletePoll.ts:84]: We should use a single query to perform this action rather than many separate queries.

**Review by jnaviask — CHANGES_REQUESTED** [review]
Why only "sorta"?

**Comment by alexyoung23j** [comment]
@jnaviask re "sorta" see the comment I left on the ticket. Tldr- the bug that was reported isn't a bug but I found a related bug

File: packages/commonwealth/server/routes/deletePoll.ts (hunk 1 above)
Comment (jnaviask) [reply packages/commonwealth/server/routes/deletePoll.ts:84]: @alexyoung23j please fix

Comment (alexyoung23j) [reply packages/commonwealth/server/routes/deletePoll.ts]: indeed

**Review by jnaviask — CHANGES_REQUESTED** [review]
Delete works but tag still not being removed until refresh.

**Comment by alexyoung23j** [comment]
@jnaviask I don't see that happening, are we talking about the poll tag that appears on the discussions page? It seems to be deleted when I navigate back to that page after removing the poll.

**Comment by jnaviask** [comment]
> @jnaviask I don't see that happening, are we talking about the poll tag that appears on the discussions page? It seems to be deleted when I navigate back to that page after removing the poll.

Yes. Navigate back to discussions via sidebar without reloading. It doesn't remove the Poll tag.

**Comment by alexyoung23j** [comment]
@jnaviask Not sure how you're seeing that, I see this: 

https://github.com/hicommonwealth/commonwealth/assets/31940965/61ece18f-68f5-4a3c-bbed-a109bb3bab77

**Review by CowMuon — APPROVED** [review]
Other than using try-catch rather than conditional logic, as we discussed, this is fine to go out. 

As discussed, 'if' conditional blocks should be when you don't exactly know what will be returned (out of a number of possibilities) and you need to check; `try` is for when you expect something to be there, and if it's not just need to handle the returned undefined or null value.

**Review by jnaviask — APPROVED** [review]

**Pull request merged**